server/src/views/frame.js

```diff
@@ -222,6 +222,7 @@ class Frame extends React.Component {
 
   clickFullPageButton(e) {
     e.preventDefault();
+    history.pushState({state: "frame"}, "", "#frame");
```

Comment:
I guess it won't be a problem if the user that comes to the page from the url with the hash sees the full page button. Maybe it will be a hint that there is something above there.